### PR TITLE
New package: wget2-2.0.1

### PR DESCRIPTION
**Pull request by dexgs**
Also included are subpackages libwget and libwget-devel.

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)

### CODE REVIEW

**Comment by dexgs** [comment]
Many of the dependencies for libwget are optional, but i have enabled them all by default as the reason someone would prefer wget2 to wget is probably the presence of these extra features.

**Comment by dexgs** [comment]
libwget gets its own package because it is licensed separately from the wget2 executable

File: srcpkgs/wget2/template [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	rm ${DESTDIR}/usr/bin/wget2_noinstall
+	install -D -m 644 -t ${DESTDIR}/usr/share/man/man3 docs/man/man3/*.3
+}
+
+libwget_package() {
+	depends="gnutls libidn2 libpsl
+	 $(vopt_if brotli 'brotli') $(vopt_if http2 'nghttp2')
+	 $(vopt_if gpgme 'libgpgme') $(vopt_if lzma 'liblzma')
+	 $(vopt_if bzip2 'bzip2') $(vopt_if zstd 'libzstd')
+	 $(vopt_if pcre 'libpcre2')"
```
Comment (oreo639) [inline srcpkgs/wget2/template:35]: Keep in mind that xbps can automatically get some of the dependencies using `objdump -p`.

Comment (dexgs) [reply srcpkgs/wget2/template:35]: which could be omitted?

Comment (dexgs) [reply srcpkgs/wget2/template:35]: I see
```
  NEEDED               libidn2.so.0
  NEEDED               libzstd.so.1
  NEEDED               libbrotlidec.so.1
  NEEDED               libz.so.1
  NEEDED               libnghttp2.so.14
  NEEDED               libpsl.so.5
  NEEDED               libgnutls.so.30
  NEEDED               libpthread.so.0
  NEEDED               libc.so.6
```
in the output of objdump -p. I guess these can be removed from the list of dependencies?

Comment (dexgs) [reply srcpkgs/wget2/template:35]: as for the libraries which aren't, do these need to stay in `depends`, or does their absence mean they were linked statically and can therefore be removed as well?

Comment (oreo639) [reply srcpkgs/wget2/template:35]: You don't need to do objdump manually to check, when building the package it lists them in `running pre-pkg hook: 04-generate-runtime-deps ...`.

File: srcpkgs/wget2/template [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+changelog="https://gitlab.com/gnuwget/wget2/-/blob/master/NEWS"
+distfiles="${GNU_SITE}/wget/wget2-${version}.tar.gz"
+checksum=0bb7fa03697bb5b8d05e1b5e15b863440826eb845874c4ffb5e32330f9845db1
+
+build_options="brotli http2 gpgme lzma bzip2 zstd pcre"
+build_options_default="brotli http2 gpgme lzma bzip2 zstd pcre"
+
+post_install() {
+	rm ${DESTDIR}/usr/bin/wget2_noinstall
+	install -D -m 644 -t ${DESTDIR}/usr/share/man/man3 docs/man/man3/*.3
```
Comment (classabbyamp) [inline srcpkgs/wget2/template:27]: use `vman` for this

File: srcpkgs/wget2/template [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ $(vopt_if gpgme 'gpgme-devel') $(vopt_if lzma 'liblzma-devel')
+ $(vopt_if bzip2 'bzip2-devel') $(vopt_if zstd 'libzstd-devel')
+ $(vopt_if pcre 'pcre2-devel')"
+depends="libwget>=${version}_${revision}"
+checkdepends="libmicrohttpd-devel gpgme"
+short_desc="GNU file and recursive website downloader"
+maintainer="chili-b <dexter.gaonshatford@gmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://gitlab.com/gnuwget/wget2"
+changelog="https://gitlab.com/gnuwget/wget2/-/blob/master/NEWS"
```
Comment (classabbyamp) [inline srcpkgs/wget2/template:18]:
```suggestion
changelog="https://gitlab.com/gnuwget/wget2/-/raw/master/NEWS"
```

File: srcpkgs/wget2/template [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		vmove "usr/lib/*.so.*"
+	}
+}
+
+libwget-devel_package() {
+	depends="libwget>=${version}_${revision}"
+	short_desc+=" - development files"
+	license="LGPL-3.0-or-later"
+	pkg_install() {
+		vmove "usr/lib/*.a"
```
Comment (classabbyamp) [inline srcpkgs/wget2/template:48]:
```suggestion
		vmove "usr/lib/*.a"
		vmove "usr/lib/*.so"
```

File: srcpkgs/wget2/template (hunk 2 above)
Comment (dexgs) [reply srcpkgs/wget2/template:27]: What is the recommended way to use vman on an entire directory?

Comment (classabbyamp) [reply srcpkgs/wget2/template:27]:
```sh
for page in docs/man/man3/*.3; do
	vman $page
done
```

**Comment by classabbyamp** [comment]
also, add the shlib to common/shlibs

File: srcpkgs/wget2/template (hunk 4 above)
Comment (dexgs) [reply srcpkgs/wget2/template:50]: why should the shared object be in the devel package?

Comment (classabbyamp) [reply srcpkgs/wget2/template:50]: so other packages can link to it. it's just a symlink to the versioned shlib

Comment (oreo639) [reply srcpkgs/wget2/template:50]: `libname.so.abiver` is used by applications at runtime.
`libname.so` is used by the compiler at compile time.

**Comment by tranzystorek-io** [comment]
Do we need to separate out all the build opts? I've had a feeling these are not usually for optional package features but for specific edge cases, e.g. `pandoc` not being available on every arch. Otherwise, there seems to be no harm in building the full feature set.

**Comment by dexgs** [comment]
> Do we need to separate the build opts?

I just tried to follow [the build options section in the manual](https://github.com/void-linux/void-packages/blob/master/Manual.md#build-options)

**Pull request closed without merging**